Classes/Parser/Expression/NodeList.php

```diff
@@ -34,25 +34,25 @@ public static function parse(Lexer $lexer): array
         while (!$lexer->isEnd()) {
             if ($lexer->isOpeningBracket()) {
                 $lexer->consume();
-
+                if ($currentText) {
+                    $contents[] = [
+                        'type' => 'text',
+                        'payload' => $currentText
+                    ];
+                }
                 if ($lexer->isForwardSlash()) {
                     $lexer->rewind();
-                    if ($currentText) {
-                        $contents[] = [
-                            'type' => 'text',
-                            'payload' => $currentText
-                        ];
-                    }
                     return $contents;
+                } elseif ($lexer->isExclamationMark()) {
```

Comment:
the `else` from `elseif` can also be ommitted here since there is a return in the if before.

Reply:
Adjusted reads much nicer as you suggested, thanks